04.ls/ls.rb

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+files = Dir.children('.').reject { |f| f.start_with?('.') }
+
+sorted = files.sort_by do |f|
+  [File.directory?(f) ? 0 : 1, f.downcase]
+end
+
+rows = (sorted.size.to_f / 3).ceil
+
+columns_array = Array.new(3) { [] }
+
+sorted.each_with_index do |file, index|
+  col = index / rows
+  columns_array[col] << file
+end
+
+(0...rows).each do |row_idx|
```

Comment:
ここは `rows.times` を使った方がより簡潔に書けますね 👍 
```suggestion
rows.times do |row_idx|
  # 各行の処理
end
```